server/controllers/StageController.js

```diff
@@ -60,6 +60,18 @@ class StageController {
     }
   }
 
+  async getStagesBySharedTripId(tripId) {
```

Comment:
U can modify getStagesByTripId

First, get the trip based on tripId, then check if isPublic or if userId matches. If either condition holds, return trip